database-factory/dbfactory.go

```diff
@@ -16,12 +16,33 @@ var templateDirPostgreSQL = "terraform/aws/database-factory-postgresql"
 func InitProvisionCluster(cluster *model.Cluster) {
 	err := ProvisionCluster(cluster)
 	if err != nil {
-		logger.WithError(err).Error("failed to deploy RDS Aurora cluster")
-		err = sendMattermostErrorNotification(cluster, err, "The Database Factory failed to deploy RDS Aurora cluster")
-		if err != nil {
-			logger.WithError(err).Error("Failed to send Mattermost error notification")
+		if isTerraformWarning(err) {
+			// Handle Terraform warning (non-fatal error)
+			logger.WithError(err).Warn("Terraform encountered a warning during deployment")
+		} else {
+			logger.WithError(err).Error("failed to deploy RDS Aurora cluster")
+			err = sendMattermostErrorNotification(cluster, err, "The Database Factory failed to deploy RDS Aurora cluster")
+			if err != nil {
+				logger.WithError(err).Error("Failed to send Mattermost error notification")
+			}
 		}
+	}
+}
 
+func InitDeleteCluster(cluster *model.Cluster) {
+	err := DeleteCluster(cluster)
+	if err != nil {
+		if isTerraformWarning(err) {
```

Comment:
this is to avoid some exit codes non-zero while is just a warning. I faced them during implementation. It is not a great way though I admit it as it is based on text.